### DISCUSSION TITLE
新特性 New Features

### DISCUSSION THREAD

**Discussion by dangfan** [canokeys/canokey-product#5] [General]
需要什么新特性，请留言，我们会仔细评估 : )

Tell us new features that you want : )

**Comment by leoleoasd** [comment]
totp浏览器插件，类似密码管理器那样

**Reply by dangfan** [reply]
浏览器跟外设通信有问题，webusb容易被占用，其他协议需要装本地程序，有什么建议吗

**Reply by z4yx** [reply]
增加一个webusb的抢占功能。在用户勾选该特性后，进入webusb时强行断开ccid（对host表现为picc not present），同时浏览器通过心跳包保持此状态。当浏览器断开后，心跳终止，ccid重新连接（picc present）。

**Reply by dangfan** [reply]
这个机制ok，要不要考虑后台跑daemon的方式呢？

**Comment by Editst** [comment]
Web Console PIV 增强一下，类似 Yubikey Manager 可以查看各个插槽信息

**Reply by dangfan** [reply]
Web Console 正在计划重写了

**Comment by sorz** [comment]
SSH resident keys

**Reply by z4yx** [reply]
这个可能是bug，我开帖跟踪下

**Reply by z4yx** [reply]
结论是当前固件不支持 FIDO 2.1，所以无法使用。见 #10

**Comment by oxalica** [comment]
换一个非蓝光的 LED 避免夜间刺眼，绿色、橙色或白色都会好很多。~~而且太亮了，能调光就更好了~~

**Comment by xiaoxi654** [comment]
触摸时输入静态密码，不限于 HOTP

**Comment by wy16W2pIilK1xgqN** [comment]
希望支持   Yubikey 的  HMAC-SHA1 Challenge-Response  协议。
该协议提供一种 挑战-响应认证，除了用于身份认证外，还可用于本地的文件的动态加密。
KeePassXC与 PasswdSafe 现在使用这个功能，作为密码库的密钥，
另外 硬件密钥 OnlyKey、VivoKey Apex 也支持这个协议。 

相关链接：
https://github.com/DangerousThings/flexsecure-ykhmac
https://keepassxc.org/docs/#faq-yubikey-howto
https://www.rfc-editor.org/rfc/rfc2104
https://www.reddit.com/r/KeePass/comments/fikgle/keepassxc_and_generally_encryption_with_an/
